src/ServerRequest.php

```diff
@@ -180,23 +180,42 @@ public static function fromGlobals()
             ->withUploadedFiles(self::normalizeFiles($_FILES));
     }
 
+    private static function extractPortFromHost($host)
+    {
+        if (preg_match('/:(\d+)$/', $host, $matches) !== 1) {
```

Comment:
I would suggest to prefix the $host with `http://` and then use `parse_url`. This way we can forward the implementation details to php itself.